src/coreclr/jit/host.h

```diff
@@ -57,7 +57,7 @@ inline FILE* procstdout()
     return stdout;
 }
 #undef stdout
-#define stdout use_jitstdout
+#define stdout jitstdout
```

Comment:
What definition of `printf` ends up picking this up? Does it work reliably with both PAL and Windows?

Reply:
I actually can't see how this would work, we have plenty of `printf`'s in JIT already that should have been erroring before if they were picking this up (due to `use_jitstdout`)

Reply:
From what I see in Release it uses stdio's printf
![image](https://user-images.githubusercontent.com/523221/188462831-3b31d8f4-071e-4889-9d04-ac16d1aac1bd.png)

Haven't checked PAL yet, building on Linux now

Reply:
> I actually can't see how this would work

Well, works for me locally 🙂

Reply:
I assume it worked just fine, it's just `stdout` was always nullptr in Release so printf used to print into void. In Debug `printf` is re-defined to use `flogf`

Reply:
> It does not work for me on Windows. The disassembly keeps going to console and nothing is written to the `JitStdOutFile` when I try this PR in release.

Yep, pushed wrong change apparently, let me check, thanks for checking

Reply:
But this works for me:
```diff
diff --git a/src/coreclr/jit/ee_il_dll.cpp b/src/coreclr/jit/ee_il_dll.cpp
index edd5fb82d63..2ab6fdb91eb 100644
--- a/src/coreclr/jit/ee_il_dll.cpp
+++ b/src/coreclr/jit/ee_il_dll.cpp
@@ -147,6 +147,14 @@ void jitShutdown(bool processIsTerminating)
     g_jitInitialized = false;
 }
 
+void jitprintf(const char* fmt, ...)
+{
+    va_list vl;
+    va_start(vl, fmt);
+    vfprintf(jitstdout, fmt, vl);
+    va_end(vl);
+}
+
 /*****************************************************************************/
 
 static CILJit g_CILJit;
diff --git a/src/coreclr/jit/host.h b/src/coreclr/jit/host.h
index 75bda4dfec9..59d2c8baca2 100644
--- a/src/coreclr/jit/host.h
+++ b/src/coreclr/jit/host.h
@@ -59,6 +59,11 @@ inline FILE* procstdout()
 #undef stdout
 #define stdout jitstdout
 
+#undef printf
+#define printf jitprintf
+
+void jitprintf(const char* fmt, ...);
+
 /*****************************************************************************/
 #endif
 /*****************************************************************************/
```

But maybe that's similar to the change you were actually testing 🙂

Reply:
haha, yeah, I've ported some bits from error.cpp (logf) to that file and it looks almost exactly like this! 😄